packages/modules/psa/api.py

```diff
@@ -0,0 +1,157 @@
+
+# these two lines enable debugging at httplib level (requests->urllib3->http.client)
+# You will see the REQUEST, including HEADERS and DATA, and RESPONSE with HEADERS but without DATA.
+# The only thing missing will be the response.body which is not logged.
+
+
+def requestPSAData(user_id: str,
+                   password: str,
+                   client_id: str,
+                   client_secret: str,
+                   manufacturer: str,
+                   soccalc: bool,
+                   vehicle_vin: str,
+                   vehicle: int) -> dict:
+
+    # Step1: Authentication: get access_token
+    scope = 'openid profile'
+    if (manufacturer == "Peugeot"):
+        brand = 'peugeot.com'
+        realm = 'clientsB2CPeugeot'
+        if (not client_id) or (not client_secret):
+            client_id = "1eebc2d5-5df3-459b-a624-20abfcf82530"
+            client_secret = "T5tP7iS0cO8sC0lA2iE2aR7gK6uE5rF3lJ8pC3nO1pR7tL8vU1"
+    elif (manufacturer == "Citroen"):
+        brand = 'citroen.com'
+        realm = 'clientsB2CCitroen'
+        if (not client_id) or (not client_secret):
+            client_id = "5364defc-80e6-447b-bec6-4af8d1542cae"
+            client_secret = "iE0cD8bB0yJ0dS6rO3nN1hI2wU7uA5xR4gP7lD6vM0oH0nS8dN"
+    elif (manufacturer == "DS"):
+        brand = 'driveds.com'
+        realm = 'clientsB2CDS'
+        if (not client_id) or (not client_secret):
+            client_id = "cbf74ee7-a303-4c3d-aba3-29f5994e2dfa"
+            client_secret = "X6bE6yQ3tH1cG5oA6aW4fS6hK0cR0aK5yN2wE4hP8vL8oW5gU3"
+    elif (manufacturer == "Opel"):
+        brand = 'opel.com'
+        realm = 'clientsB2COpel'
+        if (not client_id) or (not client_secret):
+            client_id = "07364655-93cb-4194-8158-6b035ac2c24c"
+            client_secret = "F2kK7lC5kF5qN7tM0wT8kE3cW1dP0wC5pI6vC0sQ5iP5cN8cJ8"
+    elif (manufacturer == "Vauxhall"):
+        brand = 'vauxhall.co.uk'
+        realm = 'clientsB2CVauxhall'
+        # don't have valid client_id and secret for Vauxhall
+
+    app_credentials = client_id + ':' + client_secret
+    encoded_app_credentials = base64.b64encode(app_credentials.encode()).decode()
+
+    data = {'realm': realm, 'grant_type': 'password', 'password': password, 'username': user_id, 'scope': scope}
+    headers = {
+        'Content-Type': 'application/x-www-form-urlencoded', 'Authorization': 'Basic %s' % encoded_app_credentials}
+    psa_config = req.get_http_session().post(
+        f'https://idpcvs.{brand}/am/oauth2/access_token', data=data, headers=headers).json()
+
+    access_token = psa_config['access_token']
+
+    # Step 2: Retrieve VINs and VIN_IDs in profile
+    data = {'client_id': client_id}
+    headers = {
+        'Accept': 'application/hal+json', 'Authorization': 'Bearer %s' % access_token, 'x-introspect-realm': realm}
+
+    vehicle_response = req.get_http_session().get(
+        f'https://api.groupe-psa.com/connectedcar/v4/user/vehicles?client_id={client_id}', headers=headers).json()
```

Comment:
"Best practice" ist die Parameter seperat an requests-lib zu geben:
```suggestion
    vehicle_response = req.get_http_session().get(
        "https://api.groupe-psa.com/connectedcar/v4/user/vehicles", params={"client_id": client_id}, headers=headers).json()
```
Denn falls nötig encodiert die requests-lib Parameter auf diese Weise falls nötig. Wahrscheinlich kommen in der `client_id` zwar keine Zeichen vor, die das nötig haben, aber falls das doch mal passieren sollte kann die Fehlersuche lange dauern.